### PR TITLE
feat(consent): add method that returns consent choices

### PR DESCRIPTION
**Pull request by birgernass**
### Description

v5.0.0 introduced the UMP SDK, which stores the consent as an encoded string on the device and doesn't provide further information. It is a common use case to act upon a user's choices though, especially as the Google Mobile Ads SDK won't serve any ads if the user doesn't at least provide consent to store and/or access information on the device. This PR aims to bridge that gap by introducing a new method to the `AdsConsent` interface called `getUserChoices` (happy to hear other suggestions).

I also went ahead and replaced the JSDocs inside `src/AdsConsent.ts` with TypeScript types. If the JSDocs serve a purpose, then let me know and I'll revert.

### Related issues

Related to https://github.com/invertase/react-native-google-mobile-ads/issues/96

### Checklist

- My change supports the following platforms;
  - [X] `Android`
  - [X] `iOS`
- My change includes tests;
  - [X] `e2e` tests added or updated in `__tests__e2e__`
- [X] I have updated TypeScript types that are affected by my change.
- This is a breaking change;
  - [X] No

### CODE REVIEW

**Comment by mikehardy** [comment]
This took a few conflicts from recent PR merges, but I want it to go in vs succumb to entropy so I since I was the one merging and blowing up PRs, I pulled this locally, rebased it to main, fixed the conflicts, then tested it out thorougly. I can say it has no regressions on android or ios emulators

Still needs a review but testing is half the battle I think - takes as much time as the review normally :-)

**Comment by mikehardy** [comment]
By the way - the consent form seems to come up correctly on my iOS simulator, which is kind of nice. Not on the Android Emulator, but seeing it pop up was cool. I had never even tried it believe it or not, because of the ominous warning in there. Do you think there is any way to expose the decode of the consent choices as a console.log at least?

File: src/AdsConsentPurposes.ts
Comment (mikehardy) [inline src/AdsConsentPurposes.ts:19]: Did these come from here somewhere? https://github.com/InteractiveAdvertisingBureau/iabtcf-es/tree/master/modules/core

It would be good to have a reference  in the comments of how these came to be as I imagine they will change over time

File: src/AdsConsentSpecialFeatures.ts
Comment (mikehardy) [inline src/AdsConsentSpecialFeatures.ts:19]: Same as the Features one - would be great to have a comment pointing to the source of these or the mechanism for coming up with them, for future maintenance

File: src/types/AdsConsent.interface.ts
Comment (mikehardy) [inline src/types/AdsConsent.interface.ts:175]: Same concern as the other two spots, this seems like an area that will need semi-frequent refresh from source to get the consent correct as it changes over time. This one actually looks pretty duplicate too - is there a way to de-duplicate these three files so the same text is only in one spot? There may not be but it looks like a don't-repeat-yourself violation

File: ios/RNGoogleMobileAds/RNGoogleMobileAdsConsentModule.m
Comment (mikehardy) [inline ios/RNGoogleMobileAds/RNGoogleMobileAdsConsentModule.m:126]: Is this key a constant from the form itself (here and in Java), or some other spot? I wonder if the key could be imported as a symbol so it wasn't just a magic string

More importantly, if we're storing and decoding persistent state on disk, is there any provision for versioning, checking for staleness, updating contents of that string if the form has changed? This may all be a function of the library, I am very new to this domain

**Review by mikehardy — CHANGES_REQUESTED** [review]
Mostly nit-picks and questions I don't see anything fundamentally wrong here, and at though there is no place in the example app where this is actually exercised, it is at least not a regression at the moment - all current functionality works. Should switch to approve quickly after short discussion

**Comment by mikehardy** [comment]
Great thanks! I just realized I should have issued a little bit of a warning: since I pushed to your branch on your fork, please make sure you git rebase from it before you start working locally again or things could get messy. If you've already done that or knew that sorry for boring you but if you haven't, or it gets weird and you need a hand just let me know :-), my goal was to be helpful and speed things up vs making it worse, hopefully that's the case 🤞

File: android/src/main/java/io/invertase/googlemobileads/ReactNativeGoogleMobileAdsConsentModule.java
Comment (birgernass) [reply android/src/main/java/io/invertase/googlemobileads/ReactNativeGoogleMobileAdsConsentModule.java:70]: When I tried react-native nightly a few weeks ago, playing around with the new architecture, I got build issues due to these typings. I didn't dig deep enough to tell what the reason was, but I figured it wouldn't hurt to change.

File: docs/common-reasons-for-ads-not-showing.mdx
Comment (birgernass) [reply docs/common-reasons-for-ads-not-showing.mdx:82]: Yes, I agree, I rebased and split the commit into two.

File: src/AdsConsent.ts
Comment (birgernass) [reply src/AdsConsent.ts:104]: I added such a log.

File: src/AdsConsentPurposes.ts
Comment (birgernass) [reply src/AdsConsentPurposes.ts:26]: I added links and a very brief description, let me know if that is not what you had in mind.

File: src/types/AdsConsent.interface.ts
Comment (birgernass) [reply src/types/AdsConsent.interface.ts:182]: Im not sure how this works with TypeScript, editor resolution and the like. Wanted to have the description showing up during code completion as well as next to their definitions. Do you have a goof idea? Happy to change.

File: ios/RNGoogleMobileAds/RNGoogleMobileAdsConsentModule.m
Comment (birgernass) [reply ios/RNGoogleMobileAds/RNGoogleMobileAdsConsentModule.m:127]: I sadly couldn't find an export. They refer to using this string inside their documentation as well, I referenced it in the code comments.

We do not interact with the TCString besides reading it from disk. Google and 3rd party vendors use the content of the string to check for available consent and other information. The UMP SDK stores it on the device and also automatically handles outdated values, e.g. if Google itself adds new vendors to their list. The consent modal will again be shown to the user.

The IAB and the Consent Framework are slow moving industry standards, generally speaking, and the key should stay relevant for quite a while.

File: android/src/main/java/io/invertase/googlemobileads/ReactNativeGoogleMobileAdsConsentModule.java
Comment (mikehardy) [reply android/src/main/java/io/invertase/googlemobileads/ReactNativeGoogleMobileAdsConsentModule.java:70]: That's as solid a defense as any. I know they are switching up some of the marshal/unmarshal types, could be an artifact of that. Anyway, works for me!

File: ios/RNGoogleMobileAds/RNGoogleMobileAdsConsentModule.m
Comment (mikehardy) [reply ios/RNGoogleMobileAds/RNGoogleMobileAdsConsentModule.m:127]: Okay, that makes sense - what I understood was "the libraries handle this", and the format does seem forward-compatible, sort of JSON-y

File: src/AdsConsentPurposes.ts
Comment (mikehardy) [reply src/AdsConsentPurposes.ts:26]: perfect thanks - the added links sprinkled throughout at least give a handle to all the context an interested dev (like myself) needs to go digging.

File: src/types/AdsConsent.interface.ts
Comment (mikehardy) [reply src/types/AdsConsent.interface.ts:182]: No better ideas if you don't have any. Better too much doc than too little, no reason to hold up on this

**Review by mikehardy — APPROVED** [review]
LGTM - everything looks completely addressed, by me read. Thanks a bunch for this, it's on "release next thing" track in my  mind

**Pull request merged**
